### PR TITLE
use jwt for auth

### PR DESCRIPTION
**Pull request by jzgit**

### CODE REVIEW

File: annotator.cfg [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
-ELASTICSEARCH_INDEX = 'annotator'
+SECRET_KEY = environ['SECRET_KEY']
+
+#ELASTICSEARCH_HOST = 'http://127.0.0.1:9200/'
+#SECRET_KEY = 'test$secret'
+
 
-AUTH_ON = False
-AUTHZ_ON = False
+AUTH_ON = True
```
Comment (kapluni) [inline annotator.cfg]: both are true?

File: annotator/auth.py [hunk 2]
```diff
@@ -10,6 +10,8 @@
 class Consumer(object):
     def __init__(self, key):
         self.key = key
+        self.secret = 'test$secret'
```
Comment (kapluni) [inline annotator/auth.py]: this should come from an environment variable, right?

File: annotator/auth.py [hunk 3]
```diff
@@ -102,7 +104,7 @@ class TokenInvalid(Exception):
 # Main auth routines
 
 def encode_token(token, secret):
```
Comment (kapluni) [inline annotator/auth.py]: it shoudn't really have to encode ever, right? probably fine ot change it anyway

File: run.py [hunk 4]
```diff
@@ -85,16 +85,26 @@ def before_request():
...
         # By default, this test application won't do full-on authentication
         # tests. Set AUTH_ON to True in the config file to enable (limited)
         # authentication testing.
         if current_app.config['AUTH_ON']:
-            g.auth = auth.Authenticator(lambda x: MockConsumer('annotateit'))
+            g.auth = auth.Authenticator(lambda x: auth.Consumer('1fcfd89abb8c718cf9eecd96bc542'))
         else:
             g.auth = MockAuthenticator()
 
+        user = g.auth.request_user(request)
```
Comment (kapluni) [inline run.py]: why did you have to add this? there is no default user support?

File: run.py [hunk 5]
```diff
@@ -85,16 +85,26 @@ def before_request():
...
-        g.user = MockUser('alice')
+        #g.user = MockUser('alice')
+
 
         # By default, this test application won't do full-on authentication
         # tests. Set AUTH_ON to True in the config file to enable (limited)
         # authentication testing.
         if current_app.config['AUTH_ON']:
-            g.auth = auth.Authenticator(lambda x: MockConsumer('annotateit'))
+            g.auth = auth.Authenticator(lambda x: auth.Consumer('1fcfd89abb8c718cf9eecd96bc542'))
```
Comment (kapluni) [inline run.py]: we may want to put this in an environment variable too, since this is a public repo

File: annotator.cfg (hunk 1 above)
Comment (jzgit) [inline annotator.cfg]: yeah they're confusingly authentication and authorization abbreviations

File: run.py (hunk 4 above)
Comment (jzgit) [inline run.py]: we parse the user from the request and then set the user on `g` which is a global request context in flask apps.  there was a line above this that was setting up the `MockUser` but we need the real user to limit which annotations we send to the client.

File: run.py (hunk 5 above)
Comment (jzgit) [inline run.py]: moved

File: annotator/auth.py (hunk 3 above)
Comment (jzgit) [inline annotator/auth.py]: yeah this was in the app to start.  theoretically we could serve the token from this app as well

File: annotator/auth.py (hunk 2 above)
Comment (jzgit) [inline annotator/auth.py]: moved

**Pull request closed without merging**